File: Dec2020/6Dec/DIStringMatch.java
/*
Problem Name: 942. DI String Match
Problem Link: https://leetcode.com/problems/di-string-match/
*/
class Solution {
    public int[] diStringMatch(String S) {
        int ans[] = new int[S.length()+1];
        int inc = 0, dec = S.length();
        for(int i=0;i<S.length();i++){
            if(S.charAt(i) == 'I')
                ans[i] = inc++;
            else
                ans[i] = dec--;
        }
        ans[ans.length-1] = inc;
        return ans;
    }
}

File: Dec2020/6Dec/DIStringMatch.cpp
/*
Problem name: 942. DI String Match
Problem link: https://leetcode.com/problems/di-string-match/
*/

class Solution {
public:
    vector<int> diStringMatch(string S) {
        vector<int> res(S.length()+1, -1);
        int counter = 0;
        for(int i=0;i<S.length();++i){
            if(S[i] == 'I'){
                res[i] = counter;
                counter++;
            }
        }
        for(int i=S.length();i>=0;--i){
            if(res[i] == -1){
                res[i] = counter;
                counter++;
            }
        }
        return res;
    }
};
